### PR TITLE
[hebao] improve sub-account design

### PR DESCRIPTION
**Pull request by dong77**

### CODE REVIEW

File: packages/hebao_v1/contracts/base/BaseSubAccount.sol [hunk 1]
```diff
@@ -139,7 +139,13 @@ contract BaseSubAccount is SubAccount
         }
     }
 
-    function tokenInterestRate(address, address, address) public view returns (int) { return 0; }
+    function tokenInterestRate(address, address, address)
```
Comment (Brechtpd) [inline packages/hebao_v1/contracts/base/BaseSubAccount.sol]: The latest parameter needs to be `uint` for the amount it seems.

File: packages/hebao_v1/contracts/iface/SubAccount.sol [hunk 2]
```diff
@@ -94,15 +94,20 @@ contract SubAccount
...
     /// @param token The token's address, use 0x0 for Ether.
     /// @param amount The amount to consider.
-    /// @return interestRate The interest rate in BIPs.
+    /// @return savingInterestRate The interest rate for funding the sub-account.
+    /// @return borrowInterestRate The interest rate for borrowing tokens
+    ///         from the sub-account.
     function tokenInterestRate (
         address wallet,
         address token,
         address amount
```
Comment (Brechtpd) [inline packages/hebao_v1/contracts/iface/SubAccount.sol]: `uint`

File: packages/hebao_v1/contracts/modules/transfers/TokenBalances.sol [hunk 3]
```diff
@@ -56,7 +56,7 @@ contract TokenBalances is BaseSubAccount, BaseModule
     function withdraw(address, address[] calldata, address, uint) external { revert("UNSUPPORTED"); }
     function tokenWithdrawalable (address, address ) public view returns (uint) { return 0; }
     function tokenDepositable (address, address) public view returns (uint ) { return 0; }
-    function tokenInterestRate(address, address ) public view returns (int) { return 0; }
+    function tokenInterestRate(address, address ) public view returns (uint128, uint128) { return (0, 0); }
```
Comment (Brechtpd) [inline packages/hebao_v1/contracts/modules/transfers/TokenBalances.sol]: Missing `amount`?

File: packages/hebao_v1/contracts/base/BaseSubAccount.sol (hunk 1 above)
Comment (yueawang) [reply packages/hebao_v1/contracts/base/BaseSubAccount.sol]: already changed in #822

Comment (dong77) [reply packages/hebao_v1/contracts/base/BaseSubAccount.sol]: OK, I'll cancel this PR then.

**Pull request closed without merging**